src/simpleImg.js

```diff
@@ -38,27 +38,36 @@ export default class SimpleImg extends React.PureComponent<Props, State> {
   }
 
   componentDidMount() {
-    const cachedImagesRefString = window.sessionStorage.getItem('__REACT_SIMPLE_IMG__');
     const { src } = this.props;
     const element = this.element.current;
 
     if (
-      cachedImagesRefString &&
       window.__REACT_SIMPLE_IMG__.disableAnimateCachedImg &&
       element
       // && element.getAttribute('data-from-server') === 'no'
     ) {
+      let cachedImagesRefString = null;
       try {
-        const cachedImagesRef = JSON.parse(cachedImagesRefString);
+        // Users with strict browser privacy settings will throw an error when
+        // attempting to use localStorage and sessionStorage.
+        cachedImagesRefString = window.sessionStorage.getItem('__REACT_SIMPLE_IMG__');
+      } catch (e) {
+        logError(`sessionStorage is disabled ${e}`);
+      }
 
-        if (cachedImagesRef[src]) {
-          this.setState({
-            isCached: true,
-          });
-          return;
+      if (cachedImagesRefString) {
+        try {
+          const cachedImagesRef = JSON.parse(cachedImagesRefString);
+
+          if (cachedImagesRef[src]) {
+            this.setState({
+              isCached: true,
+            });
+            return;
+          }
+        } catch (e) {
+          logError(`JSON parsing is broken ${e}`);
```

Comment:
let's combine those two if we can as well with a generic error message :)

Reply:
done!